common/predictive-text/worker/model-compositor.ts

```diff
@@ -418,9 +416,22 @@ class ModelCompositor {
   }
 
   private wordbreak(context: Context): string {
-    let wordbreaker = this.lexicalModel.wordbreaker || wordBreakers.default;
+    let model = this.lexicalModel;
 
-    return models.wordbreak(wordbreaker, context);
+    if(model.wordbreaker || !model.wordbreak) {
+      // We don't need a 12.0 / 13.0 compatibility mode here.
+      // We're either relying on defaults or on the 14.0+ wordbreaker spec.
+      let wordbreaker = model.wordbreaker || wordBreakers.default;
+
+      return models.wordbreak(wordbreaker, context);
+    } else {
+      // 1.  This model does not provide a model following the 14.0+ wordbreaking spec
+      // 2.  This model DOES define a custom wordbreaker following the 12.0-13.0 spec.
+
+      // Since the model relies on custom wordbreaking behavior, we need to use the
+      // old, deprecated wordbreaking pattern.
+      return model.wordbreak(context);
+    }
```

Comment:
This is nice and clear. It's a bit of a shame that the names `wordbreak` and `wordbreaker` are so similar -- potential for confusion -- but I am okay with it 😉